src/lib/packager/product_patterns.rb

```diff
@@ -0,0 +1,146 @@
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2016 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# ------------------------------------------------------------------------------
+#
+
+require "yast"
+
+module Yast
+
+  # Evaluate the default patterns for the currently selected products
+  class ProductPatterns
+    include Yast::Logger
+
+    def initialize
+      Yast.import "Pkg"
+    end
+
+    # Find the default patterns for all selected products.
+    # @note The package management and the products needs to be initialized and
+    #   selected *before* using this method.
+    # @return [Array<String>] pattern names
+    def names
```

Comment:
Actually I'm not sure if that would be possible, maybe it would be a bit tricky... Anyway it's not only about the initialization, also the correct products need to selected to install. So I'd leave the responsibility on the caller to properly set all what's needed.